From 9369dd1abc6f895f21bd1e06cada9563217fb98c Mon Sep 17 00:00:00 2001
From: Shivani Soni <quic_soni@quicinc.com>
Date: Thu, 2 Mar 2023 10:08:24 +0530
Subject: [PATCH] arm/arm64: Check if netdevice is bond MLO

API support to check if netdevice is bond MLO or not

Change-Id: I0b1829d6c7a46e3ef92d84058653df1b6d720363
Signed-off-by: Shivani Soni <quic_soni@quicinc.com>
---
 include/net/bonding.h | 13 +++++++++++++
 1 file changed, 13 insertions(+)

diff --git a/include/net/bonding.h b/include/net/bonding.h
index fd43733cb6c4..b30ddc0b753d 100644
--- a/include/net/bonding.h
+++ b/include/net/bonding.h
@@ -286,6 +286,19 @@ struct bond_vlan_tag {
 
 bool bond_sk_check(struct bonding *bond);
 
+/**
+ * Returns False if the net_device is not MLO bond netdvice
+ *
+ */
+static inline bool bond_is_mlo_device(struct net_device *bond_dev)
+{
+	struct bonding *bond = netdev_priv(bond_dev);
+	if (BOND_MODE(bond) == BOND_MODE_MLO)
+		return true;
+
+	return false;
+}
+
 /**
  * Returns NULL if the net_device does not belong to any of the bond's slaves
  *
-- 
2.34.1

